src/controllers/blocks/BlocksExtrinsicsController.ts

```diff
@@ -62,10 +66,23 @@ export default class BlocksExtrinsicsController extends AbstractController<Block
 			paraId: undefined,
 		};
 
+		const cacheKey =
+			hash.toString() +
+			Number(eventDocs) +
+			Number(extrinsicDocs) +
```

Comment:
```suggestion
			Number(options.extrinsicDocs) +
```
Same here. Otherwise, `Number(extrinsicDocs)` always evaluates to `NaN` in the cacheKey, whether `extrinsicDocs` is true or false.